src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -10,6 +10,15 @@ import type { AbstractDialect } from './index.js';
 
 export type TableNameOrModel = TableName | ModelStatic;
 
+// keep REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS updated when modifying this
+export type RemoveIndexQueryOptions = {
```

Comment:
Prefer an interface for this type :) (for consistency, though we have added a few that are incorrect that I should fix)

Reply:
The reason for type is when I used an interface, the rejectInvalidOptions has a type error for the options parameter:
`Argument of type 'RemoveIndexQueryOptions' is not assignable to parameter of type 'Record<string, unknown>'.
  Index signature for type 'string' is missing in type 'RemoveIndexQueryOptions'.`

Reply:
Let me take a quick look :)

Reply:
I made a comment about this; https://github.com/sequelize/sequelize/pull/15406#discussion_r1044256126

Reply:
I see, there is a bug in `rejectInvalidOptions`. `receivedOptions` should be `object` not `Record`

Reply:
@WikiRik I've tried your suggestion below however I still get the same error.
@ephys solution works, do you want me to include the change for this in this PR or in a separate one?